### DISCUSSION TITLE
bootstrap-vcpkg fails to build

### DISCUSSION THREAD

**Discussion by bclay1297** [pelicanmapping/osgearth#2627] [Support Q&A] [answered]
I am using Visual Studio 2022 on Windows 10.  I download osgEarth today and ran the bootstrap-vcpkg.bat files.  After several minutes of building the process failed.  I could not find any osgEarth libraries or executables.  The error message is shown below. there is a BitMap.cpp file in C:\.ezvcpkg\2024.07.12\buildtrees\lerc\src\js_v4.0.4-d14c4168cc.clean\src\LercLib

Error Message

more text above this line 

-- Found meshoptimizer
-- Creating H:/build/build_include/osgEarth/BuildConfig
-- Creating H:/build/build_include/osgEarth/Version
-- Configuring done (3641.5s)
CMake Error at C:/.ezvcpkg/2024.07.12/scripts/buildsystems/vcpkg.cmake:637 (_add_library):
  Cannot find source file:

    ../../third_party/lerc/src/LercLib/BitMask.cpp
Call Stack (most recent call first):
  cmake/osgearth-macros.cmake:31 (add_library)
  src/osgEarthDrivers/lerc/CMakeLists.txt:3 (add_osgearth_plugin)

CMake Error at C:/.ezvcpkg/2024.07.12/scripts/buildsystems/vcpkg.cmake:637 (_add_library):
  No SOURCES given to target: osgdb_lerc
Call Stack (most recent call first):
  cmake/osgearth-macros.cmake:31 (add_library)
  src/osgEarthDrivers/lerc/CMakeLists.txt:3 (add_osgearth_plugin)

CMake Generate step failed.  Build files cannot be regenerated correctly.

**Comment by plevy** [comment]
In CMakeLists.txt line 114, it should be trying to update the submodules, did you see:
-- Submodule update

also take a look in src/third_party/lerc or see if there is a lerc folder

**Comment by bclay1297** [comment]
Paul:
  Thanks for your reply.

The line you asked about does not exist.  Starting at line 111 the code is
as follows

# Update git submodules
# https://cliutils.gitlab.io/modern-cmake/chapters/projects/submodule.html
find_package(Git QUIET)
if(GIT_FOUND AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
    message(STATUS "Submodule update")
    execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init
--recursive
                    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
                    RESULT_VARIABLE GIT_SUBMOD_RESULT)
    if(NOT GIT_SUBMOD_RESULT EQUAL "0")
        message(FATAL_ERROR "git submodule update --init --recursive failed
with ${GIT_SUBMOD_RESULT}, please checkout submodules")
    endif()
endif()


The folder C:\osgearth\src\third_party\lerc does exist but it is empty.

I copied lerclib to C:\osgearth\src\third_party\lerc\src to see if that
helped.  It mo0ved past that error but now error on
       Cannot find source file:
../../third_party/lerc/src/LercLib/Lerc1Decode/BitStuffer.cpp

Bruce

**Comment by gwaldron** [comment, marked as answer]
Make sure when you cloned the repository you used `--recurse-submodules`:
```
git clone --recurse-submodules https://github.com/gwaldron/osgearth.git osgearth
```
